### PR TITLE
Unmount more than one mapping of same source

### PR DESCRIPTION
**Pull request by rhvgoyal**
So far this logic assumes that for every source there is only one
destination mount inside container. But there have been cases where
same source can appear at multiple destinations. For example.

docker run -ti -v /:/rootfs -v /var/lib/docker/:/var/lib/docker fedora bash

Here source "/var/lib/docker/devicemapper" will appear at two locations
in container. Frist at /var/lib/docker/devicemapper and then at
/rootfs/var/lib/docker/devicemapper.

Enhance the logic to be able to deal with multiple mappings and unmount
all of them.

Signed-off-by: Vivek Goyal <vgoyal@redhat.com>

### CODE REVIEW

**Comment by rhvgoyal** [comment]
cc @rhatdan @mrunalp

**Comment by rhatdan** [comment]
LGTM

File: src/oci-umount.c [hunk 1]
```diff
@@ -226,13 +215,60 @@ static bool is_mounted(char *path, struct mount_info *mnt_table, size_t table_sz
...
+	unsigned i;
+	char path[PATH_MAX];
+
+	for (i = 0; i < config_mounts_len; i++) {
+		if (strcmp(host_mnt, config_mounts[i].source))
+			continue;
+
+		dest = config_mounts[i].destination;
+		if ((strlen(dest) + strlen(suffix) + 1 > PATH_MAX)) {
+			pr_perror("Mapped destination=%s and suffix=%s together are longer than PATH_MAX\n", dest, suffix);
```
Comment (mrunalp) [inline src/oci-umount.c]: Can't we just log and ignore this case?

File: src/oci-umount.c [hunk 2]
```diff
@@ -226,13 +215,60 @@ static bool is_mounted(char *path, struct mount_info *mnt_table, size_t table_sz
...
+static int map_mount_host_to_container(const struct config_mount_info *config_mounts, unsigned config_mounts_len, char *host_mnt, char **cont_mnt, unsigned max_mapped)
 {
-	char *str, *rem_host_mnt;
+	char *str;
 	_cleanup_free_ char *host_mnt_dup = NULL;
-	char *destination = NULL;
-	int dest_len;
+	char *suffix = NULL;
+	int ret;
+	unsigned nr_mapped= 0;
```
Comment (mrunalp) [inline src/oci-umount.c]: nit: Formatting space before  =

File: src/oci-umount.c (hunk 1 above)
Comment (rhvgoyal) [inline src/oci-umount.c]: Ok, I will log it and skip this mount and continue.

File: src/oci-umount.c (hunk 2 above)
Comment (rhvgoyal) [inline src/oci-umount.c]: will fix.

**Comment by rhvgoyal** [comment]
@mrunalp Took care of your comments. PTAL

**Comment by mrunalp** [comment]
LGTM

**Comment by rhvgoyal** [comment]
@rhatdan can you merge this PR now.

**Pull request merged**